labman/gui/handlers/process_handlers/normalization_process.py

```diff
@@ -64,16 +64,11 @@ def post(self):
         self.write({'processes': processes})
 
 
-class DownloadNormalizationProcessHandler(BaseHandler):
+class DownloadNormalizationProcessHandler(BaseDownloadHandler):
     @authenticated
     def get(self, process_id):
         process = NormalizationProcess(int(process_id))
         text = process.generate_echo_picklist()
-
-        self.set_header('Content-Type', 'text/csv')
-        self.set_header('Expires', '0')
-        self.set_header('Cache-Control', 'no-cache')
-        self.set_header('Content-Disposition', 'attachment; filename='
-                        'NormalizationSheet_%s.txt' % process_id)
-        self.write(text)
-        self.finish()
+        compressed_plate_name = process.compressed_plate.external_id
```

Comment:
This is within the `DownloadNormalizationProcessHandler`, and does not use `process.nornalization_process`. In the prior file, it is `DownloadLibraryPrepShotgunHandler`, and it makes a very similar call but does use `process.normalization_process`. Is this correct?

Reply:
Yes, for the rather esoteric reason that the `process` variable holds an instance of a different subclass of the Process class in each of those two places :)  In this location (`DownloadNormalizationProcessHandler`), `process` IS a `NormalizationProcess` object--which doesn't have a `normalization_process` attribute since, well, it already is one ... we can therefore go straight to looking at the `compressed_plate` attribute to find out on what gdna plate this `NormalizationProcess` was performed. However, in `DownloadLibraryPrepShotgunProcessHandler`, `process` is instead a `LibraryPrepShotgunProcess` object.  A shotgun library prep process is predicated upon an upstream normalization process, so it has a `normalization_process` attribute, and we have to dereference that attribute in order get a to `NormalizationProcess` object that we can interrogate for its  `compressed_plate` attribute.  Does this clarify?

Reply:
Ah okay, thank you for the clarification!